### PR TITLE
Refactoring of animator

### PR DESCRIPTION
**Pull request by EgorSobko**
An attempt to make this control reusable.

### CODE REVIEW

File: GuillotineMenu/GuillotineMenuTransitionAnimation.swift [hunk 1]
```diff
@@ -8,159 +8,237 @@
...
 
 @objc protocol GuillotineAnimationDelegate: NSObjectProtocol {
-    // Your menu view controller will be notified about the animation progress if conforms GuillotineAnimationDelegate protocol
-    optional func menuDidCollideWithBoundary()
-    optional func menuDidFinishPresentation()
-    optional func menuDidFinishDismissal()
-    
-    optional func willStartPresentation()
-    optional func willStartDismissal()
+    optional func animatorDidFinishPresentation(animator: GuillotineTransitionAnimation)
```
Comment (maksym-letiushov) [inline GuillotineMenu/GuillotineMenuTransitionAnimation.swift:21]: thanks

File: GuillotineMenu/GuillotineMenuTransitionAnimation.swift [hunk 2]
```diff
@@ -8,159 +8,237 @@
...
+        var fromX: CGFloat
+        var fromY:CGFloat
+        var toX: CGFloat
+        var toY:CGFloat
         if self.mode == .Presentation {
-            showHostTitleLabel(false, animated: true)
+            if supportView != nil {
+                showHostTitleLabel(false, animated: true)
+            }
             view.transform = CGAffineTransformRotate(CGAffineTransformIdentity, (initialMenuRotationAngle / 180.0) * CGFloat(M_PI));
```
Comment (maksym-letiushov) [inline GuillotineMenu/GuillotineMenuTransitionAnimation.swift]: replace some code to the method degreesToRadians

File: GuillotineMenu/GuillotineMenuTransitionAnimation.swift [hunk 3]
```diff
@@ -8,159 +8,237 @@
...
-        let hostTitleLabel = UILabel()
-        menu.view.addSubview(hostTitleLabel)
-        hostTitleLabel.numberOfLines = 1;
-        
-        if let menuProt = menu as? protocol<GuillotineAnimationProtocol> {
-            hostTitleLabel.text = menuProt.hostTitle() as String
+        if let guillotineMenu = menu as? GuillotineMenu {
+            guard guillotineMenu.titleLabel != nil else { return }
+            guillotineMenu.titleLabel!.center = CGPointMake(CGRectGetHeight(supportView!.frame) / 2, CGRectGetWidth(supportView!.frame) / 2)
+            guillotineMenu.titleLabel!.transform = CGAffineTransformMakeRotation( ( 90 * CGFloat(M_PI) ) / 180 );
```
Comment (maksym-letiushov) [inline GuillotineMenu/GuillotineMenuTransitionAnimation.swift]: Use degreesToRadians method. Thanks.

File: GuillotineMenu/GuillotineMenuTransitionAnimation.swift [hunk 4]
```diff
@@ -8,159 +8,237 @@
...
+        displayLink = CADisplayLink(target: self, selector: "updateContainerMenuButton")
+        displayLink.addToRunLoop(NSRunLoop.currentRunLoop(), forMode: NSRunLoopCommonModes)
+        displayLink.paused = true
+    }
+    
+    private func degreesToRadians(degrees: CGFloat) -> CGFloat {
+        return degrees / 180.0 * CGFloat(M_PI)
+    }
+    
+    private func updateContainerMenuButton() {
```
Comment (maksym-letiushov) [inline GuillotineMenu/GuillotineMenuTransitionAnimation.swift]: This method is used as a selector so you should mark it with @objc

File: GuillotineMenu/GuillotineMenuTransitionAnimation.swift [hunk 5]
```diff
@@ -8,159 +8,237 @@
...
+        let rotationTransform: CATransform3D = menu.view.layer.presentationLayer()!.transform
+        var angle: CGFloat = 0
+        if (rotationTransform.m11 < 0.0) {
+            angle = 180.0 - (asin(rotationTransform.m12) * 180.0 / CGFloat(M_PI))
```
Comment (maksym-letiushov) [inline GuillotineMenu/GuillotineMenuTransitionAnimation.swift]: Please use radiansToDegrees method. Thanks.

File: GuillotineMenu/GuillotineMenuTransitionAnimation.swift [hunk 6]
```diff
@@ -8,159 +8,237 @@
...
         } else {
-            hostTitleLabel.alpha = 0;
+            angle = asin(rotationTransform.m12) * 180.0 / CGFloat(M_PI)
```
Comment (maksym-letiushov) [inline GuillotineMenu/GuillotineMenuTransitionAnimation.swift]: The same as above.

**Comment by maksym-letiushov** [comment]
@EgorSobko I added one commit to this branch. Please review the code ffc8a28. Address some minor comments and good to merge when you update the README file. We should provide new documentation since we drastically changed the approach hot to use and customize component. Thanks.

**Comment by maksym-letiushov** [comment]
@EgorSobko please update GuillotineMenu.podspec for version 2.0.0 and tag 2.0.0

**Comment by EgorSobko** [comment]
@maximletushov Actually, we don't have podspec at all

**Pull request merged**